migrations/20231210123554_init.sql

```diff
@@ -0,0 +1,314 @@
+-- Enums
+
+CREATE TYPE gender                  AS ENUM ('male', 'female');
+CREATE TYPE role                    AS ENUM ('user', 'admin');
+CREATE TYPE status                  AS ENUM ('available', 'unavailable');
+CREATE TYPE association             AS ENUM ('sponsor', 'organization', 'media');
+CREATE TYPE task_priority           AS ENUM ('low', 'medium', 'high');
+CREATE TYPE acceptance_status       AS ENUM ('pending', 'accepted', 'rejected');
+CREATE TYPE employee_level          AS ENUM ('basic', 'organizer', 'manager');
+CREATE TYPE employee_contract       AS ENUM ('full-time', 'part-time', 'temporary');
+CREATE TYPE event_role              AS ENUM ('worker', 'staff');
+
+
+-- Constraints
+
+CREATE DOMAIN ufloat AS float CHECK(VALUE >= 0.0);
+-- max hours per month: 24.0 * 31.0 = 744.0
+CREATE DOMAIN hours_per_month_float AS ufloat CHECK(VALUE <= 744.0);
+CREATE DOMAIN hours_per_day_float AS ufloat CHECK(VALUE <= 24.0);
+
+
+-- Tables
+
+-- user is a reserved keyword in postgresql,
+-- so we have to use double quotes
+CREATE TABLE "user"
```

Comment:
I feel like this inconsistency will be quite annoying during the DB repository development phase. Maybe the table could be called user_record instead, so w don't have wild "" in one table name. 
Just food for thought.